sgt-back-end/server.ts

```diff
@@ -0,0 +1,141 @@
+import pg from 'pg';
+import express from 'express';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+
+const app = express();
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+// type Grade = {
+//   course: string;
+//   createdAt: string;
+//   gradeId: number;
+//   name: string;
+//   score: number;
+// };
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades";
+    `;
+
+    const grades = await db.query(sql);
+    res.json(grades.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    if (name === undefined) {
+      throw new ClientError(400, 'name is a required field');
+    }
+    if (course === undefined) {
+      throw new ClientError(400, 'course is a required field');
+    }
+    if (score === undefined) {
+      throw new ClientError(400, 'score is a required field');
+    }
+    if (isNaN(score) || score > 100 || score < 0) {
+      throw new ClientError(400, 'score should be a value from 0-100');
+    }
+    const values = [`${name}`, `${course}`, `${score}`];
```

Comment:
You don't want to do it this way. What data types are in your array? (Hint: they are all strings. Why?)
You want to pass the actual data in the params and let `pg` pass the proper value to postgres.